packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -2363,4 +2464,284 @@ describe('Table row selection', () => {
             });
         });
     });
+
+    describe('with data hierarchy', () => {
+        let element: Table<SimpleTableRecord>;
+        let connect: () => Promise<void>;
+        let disconnect: () => Promise<void>;
+        let pageObject: TablePageObject<SimpleTableRecord>;
+        let selectionChangeListener: {
+            promise: Promise<void>,
+            spy: jasmine.Spy
+        };
+        let column1: TableColumnText;
+
+        function getUnsortedExpandedIndexOfRecord(id: string): number {
+            return hierarchicalData.findIndex(x => x.id === id);
+        }
+
+        beforeEach(async () => {
+            ({ element, connect, disconnect } = await setup());
+            element.idFieldName = 'id';
+            pageObject = new TablePageObject<SimpleTableRecord>(element);
+            column1 = element.querySelector<TableColumnText>('#column1')!;
+            element.parentIdFieldName = 'parentId';
+            selectionChangeListener = createEventListener(
+                element,
+                'selection-change'
+            );
+
+            await connect();
+            await element.setData(hierarchicalData);
+            await waitForUpdatesAsync();
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        describe('with single selection', () => {
+            beforeEach(async () => {
+                element.selectionMode = TableRowSelectionMode.single;
+                await waitForUpdatesAsync();
+            });
+
+            it('can programmatically select parent row', async () => {
+                await element.setSelectedRecordIds(['0']);
+                const currentSelection = await element.getSelectedRecordIds();
+                expect(currentSelection).toEqual(['0']);
+            });
+
+            it('can programmatically select leaf row', async () => {
+                await element.setSelectedRecordIds(['0.0.0']);
+                const currentSelection = await element.getSelectedRecordIds();
+                expect(currentSelection).toEqual(['0.0.0']);
+            });
+
+            it('can interactively select parent row', async () => {
+                const rowIndex = getUnsortedExpandedIndexOfRecord('0');
+                await pageObject.clickRow(rowIndex);
+                const currentSelection = await element.getSelectedRecordIds();
+                expect(currentSelection).toEqual(['0']);
+            });
+
+            it('can interactively select child row', async () => {
+                const rowIndex = getUnsortedExpandedIndexOfRecord('2.0');
+                await pageObject.clickRow(rowIndex);
+                const currentSelection = await element.getSelectedRecordIds();
+                expect(currentSelection).toEqual(['2.0']);
+            });
+        });
+
+        describe('with multi-selection', () => {
```

Comment:
Could you add/update tests to validate the "select all rows" button:
 - should show indeterminate state if all top level rows are selected but some leafs aren't
 - should select/deselect all rows in the hierarchy when clicked